src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.tsx

```diff
@@ -164,6 +165,23 @@ function ContactMethodDetailsPage({route}: ContactMethodDetailsPageProps) {
         Navigation.goBack(ROUTES.SETTINGS_CONTACT_METHODS.getRoute(backTo));
     }, [prevValidatedDate, loginData?.validatedDate, isDefaultContactMethod, backTo]);
 
+    const clearError = useCallback(() => {
+        // When removing unverified contact methods, the ValidateCodeActionForm unmounts and triggers clearError.
+        // This causes loginData to become an object, which makes sendValidateCode trigger, so we add this check to prevent clearing the error.
+        if (!loginData?.partnerUserID) {
+            return;
+        }
+        clearContactMethodErrors(contactMethod, !isEmptyObject(validateLoginError) ? 'validateLogin' : 'validateCodeSent');
+    }, [contactMethod, loginData?.partnerUserID, validateLoginError]);
+
+    useEffect(() => {
+        if (isLoadingOnyxValues || didClearError.current) {
+            return;
+        }
+        didClearError.current = true;
+        clearError();
+    }, [clearError, isLoadingOnyxValues]);
```

Comment:
```suggestion
    }, []);
```
Only on mount won't be enough here?